tests/test_subclass_contracts.py

```diff
@@ -29,19 +33,53 @@ class Developer(Employee):
     Representation Invariants:
     - self.preferred_language == "Python" or self.preferred_language == "Java"
     """
+    preferred_language: str
 
     def __init__(self, name, wage, preferred_language):
         Employee.__init__(self, name, wage)
         self.preferred_language = preferred_language
 
 
+class Teacher(Employee):
+    """
+    Represents a restaurant teacher
+
+    Representation Invariants:
+    - self.wage == self.wage_per_class * len(self.currently_teaching)
+    """
+    currently_teaching: List[str]
+    wage_per_class: float
+
+    def __init__(self, name, wage_per_class, currently_teaching):
+        Employee.__init__(self, name, wage_per_class * len(currently_teaching))
+        self.wage_per_class = wage_per_class
+        self.currently_teaching = currently_teaching
+
+    def teach_new_classes(self, currently_teaching):
```

Comment:
This is a good test, but you should also include another version with the opposite execution order: `self.change_wages` is called first, then `self.currently_teaching` is reassigned. This is what I meant by my earlier comment: `self.change_wages` will violate a representation invariant for `Teacher` but not `Employee`, and so no error should be raised in this case either.